src/commands.ts

```diff
@@ -47,6 +47,62 @@ import {
 import { PRNode } from './view/treeNodes/pullRequestNode';
 import { RepositoryChangesNode } from './view/treeNodes/repositoryChangesNode';
 
+// Modal dialog options for handling uncommitted changes during PR checkout
+const STASH_CHANGES = vscode.l10n.t('Stash changes');
+const DISCARD_CHANGES = vscode.l10n.t('Discard changes');
+
+/**
+ * Shows a modal dialog when there are uncommitted changes during PR checkout
+ * @param repository The git repository with uncommitted changes
+ * @returns Promise<boolean> true if user chose to proceed (after staging/discarding), false if cancelled
+ */
+async function handleUncommittedChanges(repository: Repository): Promise<boolean> {
+	const hasWorkingTreeChanges = repository.state.workingTreeChanges.length > 0;
+	const hasIndexChanges = repository.state.indexChanges.length > 0;
+
+	if (!hasWorkingTreeChanges && !hasIndexChanges) {
+		return true; // No uncommitted changes, proceed
+	}
+
+	const modalResult = await vscode.window.showInformationMessage(
+		vscode.l10n.t('You have uncommitted changes that would be overwritten by checking out this pull request.'),
+		{
+			modal: true,
+			detail: vscode.l10n.t('Choose how to handle your uncommitted changes before checking out the pull request.'),
+		},
+		STASH_CHANGES,
+		DISCARD_CHANGES,
+	);
+
+	if (!modalResult) {
+		return false; // User cancelled
+	}
+
+	try {
+		if (modalResult === STASH_CHANGES) {
+			// Stash all changes (working tree changes + any unstaged changes)
+			const allChangedFiles = [
+				...repository.state.workingTreeChanges.map(change => change.uri.fsPath),
+				...repository.state.indexChanges.map(change => change.uri.fsPath),
+			];
+			if (allChangedFiles.length > 0) {
+				await repository.add(allChangedFiles);
+				await vscode.commands.executeCommand('git.stash', repository);
```

Comment:
Lad exposed stash methods in the API in the last year, we should use those instead: 

https://github.com/microsoft/vscode/blob/bb42ba975ccc5df4deecb673d089f5904d61720d/extensions/git/src/api/git.d.ts#L274-L276